stdlib/cmake/modules/AddSwiftStdlib.cmake

```diff
@@ -2341,23 +2378,19 @@ function(add_swift_target_library name)
 
       if(NOT SWIFTLIB_OBJECT_LIBRARY)
         # Add dependencies on the (not-yet-created) custom lipo target.
-        foreach(DEP ${SWIFTLIB_LINK_LIBRARIES})
-          if (NOT "${DEP}" STREQUAL "icucore" AND
-              NOT "${DEP}" STREQUAL "dispatch" AND
-              NOT "${DEP}" STREQUAL "BlocksRuntime")
+        foreach(dep ${SWIFTLIB_LINK_LIBRARIES})
+          if (NOT "${dep}" MATCHES "^(icucore|dispatch|BlocksRuntime)($|-.*)$")
```

Comment:
Personally I find the single line version easier to follow (and it only does one match instead of three), but not changing it does result in a smaller diff, so let's leave it.